core/modules/system/system.admin.inc

```diff
@@ -2418,3 +2455,163 @@ function system_date_time_lookup($form, &$form_state) {
+}
+
+/**
+ * Form submit function; retroactively transliterates existing file names.
+ *
+ * @see system_transliteration_retroactive()
+ */
+function system_transliteration_retroactive_submit($form, &$form_state) {
+  include_once BACKDROP_ROOT . '/core/includes/transliteration.inc';
+  $count = 0;
+  $errors = array();
+  $result = system_transliteration_file_query()->execute();
+
+  while ($file = $result->fetchObject()) {
+    $wrapper = file_stream_wrapper_get_instance_by_uri($file->uri);
+    $scheme = file_uri_scheme($file->uri);
+
+    // Missing implementation.
+    if (!$wrapper) {
+      $errors[] = file_uri_target($file->uri);
+      continue;
+    }
+
+    // Skip non-writable stream wrappers.
+    $writeable_stream_wrappers = file_get_stream_wrappers(STREAM_WRAPPERS_WRITE);
+    if (!isset($writeable_stream_wrappers[$scheme])) {
+      continue;
+    }
+
+    // Missing file.
+    if (!file_exists($wrapper->realpath())) {
+      $errors[] = file_uri_target($file->uri);
+      continue;
+    }
+
+    // Sanitize file name.
+    $filename = transliteration_clean_filename(basename($file->uri));
+    // Build destination URI.
+    $destination = file_stream_wrapper_uri_normalize(backdrop_dirname($file->uri) . '/' . $filename);
+    $destination = file_destination($destination, FILE_EXISTS_RENAME);
+
+    // Rename and update the file record accordingly.
+    if ($wrapper->rename($file->uri, $destination)) {
+      // If possible, add a url redirect to handle old URL references.
+      if (module_exists('redirect')) {
+        $redirect = new stdClass();
+        redirect_object_prepare($redirect, array(
+          'source' => $wrapper->getDirectoryPath() . '/' . file_uri_target($file->uri),
+          'redirect' => $wrapper->getDirectoryPath() . '/' . file_uri_target($destination),
+          'status_code' => 301)
+        );
+        redirect_save($redirect);
+      }
+      $file->uri = $destination;
+      // Don't use file_save() as it modifies the timestamp.
+      backdrop_write_record('file_managed', $file, 'fid');
+      // Inform modules that the file has been updated.
+      module_invoke_all('file_update', $file);
+      $count++;
+    }
+    else {
+      $errors[] = file_uri_target($file->uri);
+    }
+  }
+
+  if ($errors) {
+    $message = t('Not all file names could be converted. The following files could not be accessed and have been ignored:');
+    $message .= theme('item_list', array('items' => $errors));
+    backdrop_set_message($message, 'error');
+  }
+  else {
+    backdrop_set_message(t('@filenames have been successfully transliterated.', array('@filenames' => format_plural($count, '1 file name', '@count file names'))));
+  }
+
+  // Flush page cache.
+  cache_clear_all();
+}
+
+/**
+ * Builds a query that returns all file names from the database containing non-ASCII characters.
+ *
+ * @return
+ *   SelectQuery
+ */
+function system_transliteration_file_query() {
```

Comment:
We may be able to eliminate this entire function, as we do no support any drivers other than MySQL. At the very least, let's remove the switch statement and just assume mysql.